src/services/inplayer.account.service.ts

```diff
@@ -237,8 +237,9 @@ export const changePasswordWithOldPassword: ChangePasswordWithOldPassword = asyn
       errors: [],
       responseData: {},
     };
-  } catch {
-    throw new Error('Failed to change password.');
+  } catch (error: unknown) {
+    const { response } = error as InPlayerError;
+    throw new Error(response.data.message);
```

Comment:
I would check if the error is really an InPlayerError instead of type asserting it.

```suggestion
  } catch (error: unknown) {
    if (error is InPlayerError) {
      throw new Error(error.response.data.message);
    }
    throw new Error('Failed to change password');
```

Chances are small, but consider this example:

```ts
try {
  throw new InPlayerError(nofn(''));
} catch(error: unknown) {
  const { response } = error as InPlayerError;
  throw new Error(response.data.message);
}
```

The `nofn` call will trigger a ReferenceError, and the catch assumes it's an `InPlayerError`.

Reply:
Yeah, good point! Thanks for the suggestion!